### PR TITLE
[MINOR][UI] Simplify columnDefs in stagepage.js

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Simplify `columnDefs` in stagepage.js

### Why are the changes needed?

Reduce hardcode in stagepage.js and potential inconsistency for hidden/show in future changes.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

Locally verified.

image

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @ulysses-you

File: core/src/main/resources/org/apache/spark/ui/static/stagepage.js [hunk 1]
```diff
@@ -1079,15 +1069,8 @@ $(document).ready(function () {
...
-            { "visible": false, "targets": 12 },
-            { "visible": false, "targets": 13 },
-            { "visible": false, "targets": 14 },
-            { "visible": false, "targets": 15 },
-            { "visible": false, "targets": 16 },
-            { "visible": false, "targets": 17 },
-            { "visible": false, "targets": 18 },
-            { "visible": false, "targets": 21 }
+            { "visible": false, "targets": optionalColumns },
+            { "visible": false, "targets": 18 }, // accumulators
```
Comment (ulysses-you) [inline core/src/main/resources/org/apache/spark/ui/static/stagepage.js:1073]: does it affect the display ordering, before is .`.., 18, 21`, now is `..., 21, 18`

Comment (yaooqinn) [reply core/src/main/resources/org/apache/spark/ui/static/stagepage.js:1073]: no

**Comment by ulysses-you** [comment]
I tested it locally with accumulator, looks fine to me

**Comment by yaooqinn** [comment]
thanks @ulysses-you for the verification

**Comment by HyukjinKwon** [comment]
cc @gengliangwang and @sarutak FYI

**Review by sarutak — APPROVED** [review]
Yeah, we can refactor like this change.
https://datatables.net/reference/option/columnDefs

**Comment by sarutak** [comment]
GA failure seems not to be relevant to this change.
Merging to `master`. Thank you @yaooqinn and all!

**Pull request closed without merging**